src/main/java/roomescape/dto/request/ReservationRequest.java

```diff
@@ -7,25 +7,29 @@ public class ReservationRequest {
 
     private static final Pattern DATE_PATTERN = Pattern.compile("^\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])$");
 
-    private final String date;
+    private final LocalDate date;
     private final String name;
     private final Long timeId;
 
-    public ReservationRequest(String date, String name, Long timeId) {
-        validateDateAndTime(date);
+    private ReservationRequest(LocalDate date, String name, Long timeId) {
         this.date = date;
         this.timeId = timeId;
         this.name = name;
     }
 
-    private void validateDateAndTime(String date) {
+    public static ReservationRequest of(String name, String date, Long timeId) {
+        validateDateAndTime(date);
+        return new ReservationRequest(LocalDate.parse(date), name, timeId);
```

Comment:
```suggestion
        LocalDate date;
        try {
            date = LocalDate.parse(originDate);
        } catch (DateTimeParseException e) {
            throw new IllegalArgumentException("날짜는 0000-00-00 형식입니다");
        }
        return new ReservationRequest(date, name, timeId);
```

이런식으로 하면 pattern을 사용하지 않을 수도 있겠네요

Reply:
DTO도 콘솔을 개발하면서 재사용을 했습니다! 그래서 팩토리 메서드 패턴을 사용하면서 검증을 진행하고 있습니다!